pallets/msa/src/lib.rs

```diff
@@ -408,22 +409,23 @@ pub mod pallet {
 		///
 		/// ### Errors
 		/// - Returns [`InvalidSelfRevoke`](Error::InvalidSelfRevoke) if `origin` and `key` are the same.
-		/// - Returns [`KeyRevoked`](Error::KeyRevoked) if `key` is already expired.
 		/// - Returns [`NotKeyOwner`](Error::NotKeyOwner) if `origin` does not own the MSA ID associated with `key`.
 		/// - Returns [`NotKeyExists`](Error::NoKeyExists) if `origin` or `key` are not associated with `origin`'s MSA ID.
-		///
-		#[pallet::weight(T::WeightInfo::revoke_msa_key())]
-		pub fn revoke_msa_key(origin: OriginFor<T>, key: T::AccountId) -> DispatchResult {
+		/// ### Remarks
+		/// - Removal of key deletes the association of the key with the MSA.
+		/// - The key can be re-added to same or another MSA if needed.
+		#[pallet::weight(T::WeightInfo::delete_msa_key())]
+		pub fn delete_msa_key(origin: OriginFor<T>, key: T::AccountId) -> DispatchResult {
 			let who = ensure_signed(origin)?;
 
 			ensure!(who != key, Error::<T>::InvalidSelfRevoke);
 
 			let who = Self::try_get_key_info(&who)?;
 			let key_info = Self::try_get_key_info(&key)?;
-			ensure!(who.expired == T::BlockNumber::zero(), Error::<T>::KeyRevoked);
 			ensure!(who.msa_id == key_info.msa_id, Error::<T>::NotKeyOwner);
 
-			Self::revoke_key(&key)?;
+			Self::delete_key_for_msa(who.msa_id, &key)?;
```

Comment:
I think we need a transaction here

Reply:
Or we can account for weights for delete_key_for_msa and delete_key_info within this extrinsic.
What you think?

Reply:
@enddynayn  I combined the two in one where key is deleted from MSAKeys as well as KeyInfo

lets talk about its transactional nature

Reply:
@enddynayn  maybe I am mistaken this function is within a transaction (extrinsic)

Reply:
After chatting with @enddynayn on this, what I believe his take to be is that both `key_list.remove(index.unwrap());` and the `KeyInfoOf::<T>::remove(key);` lines in `delete_key_for_msa` should happen like a single DB transaction would. Atomic in the sense that if there is an error when removing the index from key_list, we should not continue with removing the key from KeyInfoOf

Reply:
I am not sure if a race condition can cause the above state to occur? Is a substrate node single threaded? Will need to see if remove on StorageMap returns any kind of an error

Reply:
I am not sure what you mean by this

Reply:
After discussing this in slack, here is what I meant to say more clearly: 
Do we need to ensure that the removal of the value in `MsaKeysOf` inside of `delete_key_for_msa` is successful before proceeding with the `KeyInfoOf::<T>::remove(key);`. If try_mutate is thread safe or if the runtime is singlethreaded we likely don't need to worry about this.